scaleway/resource_tem_domain.go

```diff
@@ -0,0 +1,219 @@
+package scaleway
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	tem "github.com/scaleway/scaleway-sdk-go/api/tem/v1alpha1"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+func resourceScalewayTemDomain() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceScalewayTemDomainCreate,
+		ReadContext:   resourceScalewayTemDomainRead,
+		DeleteContext: resourceScalewayTemDomainDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Timeouts: &schema.ResourceTimeout{
+			Create:  schema.DefaultTimeout(defaultTemDomainTimeout),
+			Read:    schema.DefaultTimeout(defaultTemDomainTimeout),
+			Delete:  schema.DefaultTimeout(defaultTemDomainTimeout),
```

Comment:
Only `TimeoutDelete` is used, they should be used or not included as it allows user to override a timeout that does not exist